### PR TITLE
Add support for negating a complex number

### LINKED ISSUE #101: RFC: complex number negation
## Checklist

> Please ensure the following tasks are completed before filing an issue.

* [x] Read and understood the [Code of Conduct][code-of-conduct].
* [x] Searched for existing issues and pull requests.
* [x] If this is a general question, searched the [FAQ][faq] for an existing answer.
* [x] If this is a feature request, the issue name begins with `RFC: `.

## Description

> Description of the issue (or feature request).

Add support for negating a complex number.

Package name: `@stdlib/math/base/complex/negate`
Alias: `cneg`

Behavior:

``` javascript
var v = cneg( re, im );
// returns [ -re, -im ]
```

The implementation is trivial; however, in ES6 envs, this is useful when combined with destructuring.

## Related Issues

> Does this issue (or feature request) have any related issues?

None.

## Questions

> Any questions for reviewers?

No.

## Other

> Any other information relevant to this issue (or feature request)? This may include screenshots, references, stack traces, sample output, and/or implementation notes.

No.

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md
[faq]: https://github.com/stdlib-js/stdlib/blob/develop/FAQ.md

### PR DESCRIPTION
**Pull request by ShraddheyaS**
Resolves #101.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing], including the relevant style guides.
-   [x] Read and understand the [Code of Conduct][code-of-conduct].
-   [x] Read and understood the [licensing terms][license].
-   [x] Searched for existing issues and pull requests **before** submitting this pull request.
-   [x] Filed an issue (or an issue already existed) **prior to** submitting this pull request.
-   [x] Rebased onto latest `develop`.
-   [x] Submitted against `develop` branch.

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds the feature of the negation of complex numbers

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #101 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

Yes.

 - The last 2 checkboxes are not ticked because I am putting in this PR against the `feature/complex-negate` branch.

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md

[license]: https://github.com/stdlib-js/stdlib/blob/develop/LICENSE

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/complex/negate/README.md [hunk 1]
```diff
@@ -46,13 +46,13 @@ Negates a `complex` number comprised of a **real** component `re` and an **imagi
 
 ```javascript
 var v = cnegate( -4.2, 5.5 );
-// returns [ TODO, TODO ]
+// returns [ 4.2, -5.5 ]
 
 v = cnegate( 0.0, 0.0 );
-// returns [ TODO, TODO ]
+// returns [ 0.0, 0.0 ]
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/complex/negate/README.md]: Is this correct? Why not negative zeros?

File: lib/node_modules/@stdlib/math/base/complex/negate/lib/cnegate.js [hunk 2]
```diff
@@ -32,24 +32,26 @@
...
+* // returns [ 4.2, -5.5 ]
 *
 * var bool = ( v === out );
 * // returns true
 *
 * @example
 * var out = new Array( 2 );
 * var v = cnegate( out, 0.0, 0.0 );
-* // returns [ TODO, TODO ]
+* // returns [ 0.0, 0.0 ]
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/complex/negate/lib/cnegate.js]: Why not negative zeros?

File: lib/node_modules/@stdlib/math/base/complex/negate/lib/index.js [hunk 3]
```diff
@@ -27,21 +27,21 @@
 * var cnegate = require( '@stdlib/math/base/complex/negate' );
 *
 * var v = cnegate( -4.2, 5.5 );
-* // returns [ TODO, TODO ]
+* // returns [ 4.2, -5.5 ]
 *
 * v = cnegate( 0.0, 0.0 );
-* // returns [ TODO, TODO ]
+* // returns [ 0.0, 0.0 ]
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/complex/negate/lib/index.js]: Why not negative zeros?

File: lib/node_modules/@stdlib/math/base/complex/negate/lib/main.js [hunk 4]
```diff
@@ -35,24 +35,24 @@ var negate = require( './cnegate.js' );
...
-* // returns [ TODO, TODO ]
+* // returns [ 4.2, -5.5 ]
 *
 * var bool = ( v === out );
 * // returns true
 *
 * @example
 * var v = cnegate( 0.0, 0.0 );
-* // returns [ TODO, TODO ]
+* // returns [ 0.0, 0.0 ]
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/complex/negate/lib/main.js]: Why not negative zeros?

File: lib/node_modules/@stdlib/math/base/complex/negate/test/test.js [hunk 5]
```diff
@@ -42,8 +42,16 @@ tape( 'the function negates real and imaginary components', function test( t ) {
...
+	actual = cnegate( -4.2, 5.5 );
+	expected = [ 4.2, -5.5 ];
+	t.deepEqual( actual, expected, 'returns expected value' );
+
+	actual = cnegate( 9.99999, 0.1 );
+	expected = [ -9.99999, -0.1 ];
+	t.deepEqual( actual, expected, 'returns expected value' );
 
+	actual = cnegate( 0.0, 0.0 );
+	expected = [ 0.0, 0.0 ];
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/complex/negate/test/test.js]: This test is a bit more nuanced, as one needs to test for negative/positive zeros. Ideally, we would have tests for both.

**Review by kgryte — CHANGES_REQUESTED** [review]
@ShraddheyaS Thanks for working on this! Main questions center around positive and negative zeros. Once these are resolved, this PR should be ready for acceptance.

File: lib/node_modules/@stdlib/math/base/complex/negate/test/test.js (hunk 5 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/complex/negate/test/test.js]: I believe this test is actually already taken care of in the tests starting at L118 and L125. Feel free to modify this test with, say, two negative numbers.

File: lib/node_modules/@stdlib/math/base/complex/negate/test/test.js [hunk 6]
```diff
@@ -51,6 +51,10 @@ tape( 'the function negates real and imaginary components', function test( t ) {
 	t.deepEqual( actual, expected, 'returns expected value' );
 
 	actual = cnegate( 0.0, 0.0 );
+	expected = [ -0.0, -0.0 ];
+	t.deepEqual( actual, expected, 'returns expected value' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/complex/negate/test/test.js]: Unfortunately, `deepEqual` does not check for sign equality when comparing zeros. Hence, the following test passes

```javascript
actual = cnegate( 0.0, 0.0 );
expected = [ 0.0, 0.0 ];
t.deepEqual( actual, expected, 'returns expected value' );
```
even if it shouldn't. Unfortunately, one cannot use `deepEqual`, but, instead, one must manually check individual array elements. In this case, that entails checking for value and sign equality. This has already been done, in later tests. So having tests here which test function signed zero behavior is not necessary. Perhaps, better might be to just test other values, perhaps with a different sign combination. For example,

```javascript
actual = cnegate( -1.0, -1.0 );
expected = [ 1.0, 1.0 ];
t.deepEqual( actual, expected, 'returns expected value' );
```

**Review by kgryte — APPROVED** [review]
@ShraddheyaS Looks good! Will merge once the canary build passes.

File: lib/node_modules/@stdlib/math/base/complex/negate/test/test.js (hunk 6 above)
Comment (ShraddheyaS) [reply lib/node_modules/@stdlib/math/base/complex/negate/test/test.js]: Yes, I realized the redundancy and changed the tests.

File: lib/node_modules/@stdlib/math/base/complex/negate/test/test.js (hunk 5 above)
Comment (ShraddheyaS) [reply lib/node_modules/@stdlib/math/base/complex/negate/test/test.js]: Yes, thank you for pointing it out. I saw the `isPositiveZero` and `isNegativeZero` tests now. I have changed these tests in the latest commit.
